x/kyllian/dust-mcp/src/cli.ts

```diff
@@ -1,31 +1,135 @@
+#!/usr/bin/env node
+
 import { Command } from "commander";
-import { authenticate } from "./auth.js";
-import { isAuthenticated } from "./config.js";
-
-const program = new Command();
-
-program.name("dust-mcp").description("Dust MCP CLI").version("1.0.0");
-
-program
-  .command("auth")
-  .description("Authenticate with Dust")
-  .action(async () => {
-    try {
-      // const alreadyAuthenticated = await isAuthenticated();
-      // if (alreadyAuthenticated) {
-      //   console.log("You are already authenticated.");
-      //   return;
-      // }
-
-      console.log("Authenticating with Dust...");
-      await authenticate();
-      console.log(
-        "Authentication successful! Your credentials have been saved."
-      );
-    } catch (error) {
-      console.error("Authentication failed:", error);
+import { getDustAPI } from "./api.js";
+import { isLoggedIn, login, logout } from "./auth.js";
+import { startServer } from "./server.js";
+import { version } from "./version.js";
+
+const loginAction = async (quiet = false) => {
```

Comment:
do we need all the try/catch in these methods ? Our API methods should return `Result` instances, so any exception is definitely unexpected (and should just bubble up and crash the process I think)